test/widgets/content_test.dart

```diff
@@ -1028,9 +1062,29 @@ void main() {
       tester.widget(find.text('λ', findRichText: true));
     });
 
-    testWidgets('fallback to displaying KaTeX source if unsupported KaTeX HTML', (tester) async {
-      await prepareContent(tester, plainContent(ContentExample.mathInlineUnknown.html));
-      tester.widget(find.text(r'\lambda'));
+    group('fallback to displaying KaTeX source if unsupported KaTeX HTML', () {
+      testContentSmoke(ContentExample.mathInlineUnknown);
+
+      assert(ContentExample.mathInlineUnknown.html.startsWith('<p>'));
+      assert(ContentExample.mathInlineUnknown.html.endsWith('</p>'));
+      final unsupportedKatexHtml = ContentExample.mathInlineUnknown.html
+        .substring(3, ContentExample.mathInlineUnknown.html.length - 4);
+
+      testFontWeight('is bold in bold span',
+        // Regression test for: https://github.com/zulip/zulip-flutter/issues/1812
+        expectedWght: 600,
+        // **$$ \lambda $$**
+        content: plainContent('<p><strong>$unsupportedKatexHtml</strong></p>'),
+        styleFinder: (tester) => mergedStyleOf(tester, r'\lambda')!,
```

Comment:
nit: can avoid the need to look up the example to see that `\lambda` is the right text to look for:
```suggestion
        styleFinder: (tester) => mergedStyleOf(tester, ContentExample.mathInlineUnknown.expectedText)!,
```

(or maybe pull that into a variable next to `unsupportedKatexHtml`)